encryptor/utils.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2019, Cossack Labs Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encryptor
+
+import (
+	"errors"
+	"github.com/cossacklabs/acra/sqlparser"
+)
+
+var errNotFoundtable = errors.New("not found table for alias")
+var errNotSupported = errors.New("not supported type of sql node")
+
+type columnInfo struct {
+	Name  string
+	Table string
+	Alias string
+}
+
+var errEmptyTableExprs = errors.New("empty table exprs")
+
+// getFirstTableWithoutAlias search table name from "FROM" expression which has not any alias
+// if more than one table specified without alias then return errNotFoundTable
+func getFirstTableWithoutAlias(fromExpr sqlparser.TableExprs) (string, error) {
```

Comment:
Here and below in this file there is a lot of parsing code with no apparent tests for it.

I figure, is it tested transitively by the integration test suite?

Reply:
You know, I looked one more time on this code, found the piece of SQL code which I used for tests when wrote it and decided to add tests too. And I found a case when it worked incorrectly)) Added query and tests for it with fix

Reply:
I know that feel :) You try to fix one thing, pick it up, and there’s a whole bunch of bugs scattering around from under it! :D

I’m glad there’s fewer of them now.